packages/envconfig/src/utils.ts

```diff
@@ -0,0 +1,155 @@
+import * as fs from 'fs';
+import { filterNullAndUndefined } from '@temporalio/common/lib/internal-workflow';
+import {
+  ClientConfigProfile,
+  ClientConfigTLS,
+  ClientConfig,
+  tomlClientConfigProfile,
+  tomlClientConfigTLS,
+  tomlClientConfig,
+  ConfigDataSource,
+} from './types';
+
+export function loadConfigData(source?: ConfigDataSource): Buffer | undefined {
+  if (!source) return undefined;
+
+  if ('path' in source) {
+    return readPathSync(source.path);
+  }
+
+  return Buffer.isBuffer(source.data) ? source.data : Buffer.from(source.data);
+}
+
+export function readPathSync(path?: string): Buffer | undefined {
+  if (path === undefined) {
+    return undefined;
+  }
+
+  return fs.readFileSync(path);
+}
```

Comment:
I'm not sure I understand the usefulness of this function. Shouldn't the caller just check the presence of `path`? It adds some confusion about control flow as opposed to just using `fs.readFileSync` directly

Reply:
Yeah fair - could probably be replaced with a ternary expression